File: pingdart-dart/lib/src/services/sms_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class SmsService {
  final http.Client client;
  final String apiKey;
  final String baseUrl;

  SmsService({required this.client, required this.apiKey, required this.baseUrl});

  Future<dynamic> sendSms(String to, String text, {String? templateId, String route = 'pingdart', String unicode = 'true'}) async {
    final response = await client.post(
      Uri.parse('${baseUrl}email/send-sms'),
      headers: {
        'x-api-key': apiKey,
        'Content-Type': 'application/json',
      },
      body: jsonEncode({
        'to': to,
        'text': text,
        'templateId': templateId,
        'route': route,
        'unicode': unicode,
      }),
    );
    return jsonDecode(response.body);
  }
}

File: pingdart-js/src/services/sms.service.js
class SmsService {
    constructor(http, config) {
        this.http = http;
    }

    async sendSMS({ to, templateId, text, route = "pingdart", unicode = "true" }) {
        try {
            const response = await this.http.post("/email/send-sms", {
                to,
                templateId,
                text,
                route,
                unicode
            });
            return response.data;
        } catch (error) {
            throw new Error(`SMS sending error: ${error.message}`);
        }
    }
}

export default SmsService;
